src/shared/HtmlReady.js

```diff
@@ -134,7 +134,7 @@ function link(state, child) {
         state.links.add(url)
         if(state.mutate) {
             // If this link is not relative, http, or https -- add https.
-            if(! /^\/(?!\/)|(https?:)?\/\//.test(url)) {
+            if(! /^((#)|(\/(?!\/))|((https?:)?\/\/))/.test(url)) {
```

Comment:
I get that you were just modifying this code, but this regular expression is atrocious. Can we perform this mutation by parsing the URL?

Reply:
In a parallel universe where I've not hit my time budget, sure. In this one. 👎  :/

Reply:
can you add test coverage for this